app/controllers/users_controller.rb

```diff
@@ -37,9 +39,13 @@ def team_valid?
   end
 
   def setup_user_in_cognito
-    temporary_password = ('a'..'z').to_a.sample(3) + ('A'..'Z').to_a.sample(3) + ('0'..'9').to_a.sample(3) + ('!'..'/').to_a.sample(1)
+    temporary_password = ""
+    until password_meets_criteria?(temporary_password) do
+      temporary_password = SecureRandom.urlsafe_base64(20)
```

Comment:
also, 20 characters might be too long? just thinking it might break over mutliple lines in the email etc... 12 should be secure enough

Reply:
Just did some tests, with 12 characters there's about 7% chance that it will not match the password requirements, meaning the loop needs to be executed multiple times. When adding a number to the random string the chance drops to 0.1% ... I would suggest then doing
`SecureRandom.urlsafe_base64(12).insert(SecureRandom.random_number(11), SecureRandom.random_number(9).to_s)`... -> it inserts a random number to a random position (idea by @j-maynard )

Reply:
Yeah, it hadn't dawned on me that the likelihood of failed attempts would be that high.  I've committed Jamie's suggestion. 👍